crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs

```diff
@@ -286,6 +288,43 @@ impl LedgerRpcProvider for LedgerDB {
         self.get_batches(&ids, query_mode)
     }
 
+    fn get_soft_batches(
+        &self,
+        soft_batch_ids: &[SoftBatchIdentifier],
+    ) -> Result<Vec<Option<SoftBatchResponse>>, anyhow::Error> {
+        anyhow::ensure!(
+            soft_batch_ids.len() <= MAX_SOFT_BATCHES_PER_REQUEST as usize,
+            "requested too many soft batches. Requested: {}. Max: {}",
+            soft_batch_ids.len(),
+            MAX_BATCHES_PER_REQUEST
+        );
+
+        let mut out = Vec::with_capacity(soft_batch_ids.len());
+        for soft_batch_id in soft_batch_ids {
+            if let Some(soft_batch) = self.get_soft_batch(soft_batch_id)? {
+                out.push(Some(soft_batch));
+            } else {
+                out.push(None);
+            }
+        }
+        Ok(out)
+    }
+
+    fn get_soft_batches_range(
+        &self,
+        start: u64,
+        end: u64,
+    ) -> Result<Vec<Option<SoftBatchResponse>>, anyhow::Error> {
+        anyhow::ensure!(start <= end, "start must be <= end");
+        anyhow::ensure!(
```

Comment:
math's wrong

start 10
end 30 
max allowed is 20

i get to take 21 soft batches lol